engine/common/follower/compliance_core.go

```diff
@@ -201,6 +201,14 @@ func (c *ComplianceCore) processCoreSeqEvents(ctx irrecoverable.SignalerContext,
 
 	doneSignal := ctx.Done()
 	for {
+		// Check if shutdown was requested before attempting to process queued events.
+		// This helps to prioritize timely reaction in case of shutdown.
+		select {
+		case <-doneSignal:
+			return
+		default:
+		}
+
```

Comment:
@durkmurder added this for more reactive shutdown, since some of the times, the test didn't manage to close the worker routine in time (don't really understand why, but it caused `pebble: closed` panics because there were still writes attempted despite the database already being closed by the test